src/Raven.Server/Documents/Indexes/Persistence/Corax/CoraxIndexPersistence.cs

```diff
@@ -111,8 +114,45 @@ public override void Clean(IndexCleanup mode)
         // lucene method
     }
 
+    private const int IndexingCompressionMaxTestDocuments = 10000;
+
+
     public override void Initialize(StorageEnvironment environment)
     {
+        var contextPool = _index._contextPool;
+
+        using (contextPool.AllocateOperationContext(out TransactionOperationContext context))
+        using (var tx = context.OpenReadTransaction())
+        {
+            // It is already initialized, we can return.
+            if (CompactTree.HasDictionary(tx.InnerTransaction.LowLevelTransaction))
+                return;
+        }
+
+        var documentStorage = _index.DocumentDatabase.DocumentsStorage;
+
+        using (CultureHelper.EnsureInvariantCulture())
+        using (contextPool.AllocateOperationContext(out TransactionOperationContext indexContext))
+        using (var queryContext = QueryOperationContext.Allocate(_index.DocumentDatabase, _index))
+        using (CurrentIndexingScope.Current = new CurrentIndexingScope(_index, documentStorage, queryContext, _index.Definition, indexContext, _index.GetOrAddSpatialField, _index._unmanagedBuffersPool))
```

Comment:
Instead of exposing `_unmanagedBuffersPool` and `GetOrAddSpatialField()` via `internal` you might add `CreateCurrentIndexingScope()` and use it in all places where we create it currently using `new`

Reply:
That's a good idea

Reply:
Good idea.